OpenKeychain/src/main/java/org/sufficientlysecure/keychain/pgp/CanonicalizedSecretKey.java

```diff
@@ -319,6 +320,28 @@ public RSAPrivateCrtKey getCrtSecretKey() throws PgpGeneralException {
         return (RSAPrivateCrtKey)retVal;
     }
 
+    // For use only in card export; returns the secret key.
```

Comment:
I understand your point, and I implemented this method in this class because of an already existing analog method called getCrtSecretKey for RSA. Both methods (getCrtSecretKey and getECSecretKey) require to have access to the private field mPrivateKey of the instance. Implementing these methods out of the class CanonicalizedSecretKey would require to un-private the field mPrivateKey, or to create an accessor for this field. These methods are only used in security token related code, but if we create a "wrapping" class, it will require to alter a lot of code to propagate the use of this new class. Tell me what you think is best and I will implement it.